src/Common/src/System/Net/Http/TlsCertificateExtensions.cs

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// Defines Extension methods which are meant to be re-used across WinHttp & UnixHttp Handlers
+    /// </summary>
+    internal static class TLSCertificateExtensions
+    {
+        private const string ClientCertificateOid = "1.3.6.1.5.5.7.3.2";
+
+        /// <summary>
+        ///   returns true if the X509 Certificate can be used  as SSL Client Certificate.
+        /// </summary>
+        internal static bool CanAuthenticateClient(this X509Certificate2 cert)
+        {
+            if (cert == null || cert.Extensions == null)
+            {
+                return false;
+            }
+
+            foreach(X509Extension extension in cert.Extensions)
+            {
+                X509EnhancedKeyUsageExtension enhancedUsageExt = extension as X509EnhancedKeyUsageExtension;
+                if (enhancedUsageExt != null)
+                {
+                    foreach(Oid oid in enhancedUsageExt.EnhancedKeyUsages)
+                    {
+                        if (string.Equals(ClientCertificateOid, oid.Value))
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+
+            return false;
```

Comment:
A certificate containing no EKU extension is defined to be valid for all extended purposes.  So really this should be more like

```C#
        {
            if (cert == null || cert.Extensions == null)
            {
                return false;
            }

            foreach(X509Extension extension in cert.Extensions)
            {
                X509EnhancedKeyUsageExtension enhancedUsageExt = extension as X509EnhancedKeyUsageExtension;
                if (enhancedUsageExt != null)
                {
                    foreach(Oid oid in enhancedUsageExt.EnhancedKeyUsages)
                    {
                        if (string.Equals(ClientCertificateOid, oid.Value))
                        {
                            return true;
                        }
                    }

                    return false;
                }
            }

            // No EKU extension was present, considered valid for client auth.
            return true;
        }
```